### PR TITLE
Add administrator role

### LINKED ISSUE #109: Add administrator role
The **administrator** role should provide the following privileges:
  - to create teams;
  - to remove _empty_ teams;
  - to create a user;
  - to update all the user information, except password;
  - to invite new users;
  - to add users to teams;
  - to remove users from teams;
  - to add holidays;
  - to update holidays;
  - to remove holidays.

### LINKED ISSUE #120: Manager should not do any manipulations on teams

### PR DESCRIPTION
**Pull request by eleidan**
The main idea is to add the 'admin' role.
See #109 for more details.

In scope of this implementation the following changes are introduced:
  - the manager role does not allow manipulations on teams;
  - user can have more than one unique role in the team;
  - only admin can work on teams;
  - BB teams implementation is dramatically changed;
  - BB App.Models.User is added;
  - BB App.Models.Role is added;
  - BB App.Collections.Users is added;
  - BB App.Collections.Roles is added;
  - RoR TeamRolesController is added.
  - templates for buttons Edit, Show, Hide, Rename, Delete

Fixes #109
Fixes #120

### CODE REVIEW

File: app/assets/javascripts/views/editable_team.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+  onRoleSuccess: function(model) {
+    this.roles.add(model);
+    this.$currentRoleButton.parent().addClass('active');
+  },
+
+  onTeamError: function(model, response, options) {
+    var message = '';
+
+    console.log('onTeamError', arguments);
```
Comment (afurm) [inline app/assets/javascripts/views/editable_team.js]: Do we need this?

File: app/assets/javascripts/views/role.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  template: JST['templates/role'],
+
+  operationsEvents: function() {
+    return {
+      'change input[name=roles]': this.onToggle,
+    };
+  },
+
+  // events: {
+  //   'change input[name=roles]': 'onToggle',
```
Comment (afurm) [inline app/assets/javascripts/views/role.js]: Same here

File: app/assets/javascripts/views/role.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    // var $input = $(event.target),
+    //     toBeAdded = $input.prop('checked'),
+    //     userID = $input.data('userId');
+    //
+    // console.log(userId);
+    // if (toBeAdded) {
+    //   role.set('role', $input.val());
+    //   role.set('user_id', userID);
+    //   role.set('team_id', this.team.get('id'));
+    //
```
Comment (afurm) [inline app/assets/javascripts/views/role.js]: See comments above.

File: app/assets/javascripts/views/editable_team.js (hunk 1 above)
Comment (eleidan) [inline app/assets/javascripts/views/editable_team.js]: Yes. Till all possible cases are covered.

Comment (afurm) [inline app/assets/javascripts/views/editable_team.js]: :ok_hand:

File: app/assets/javascripts/views/role.js (hunk 2 above)
Comment (eleidan) [inline app/assets/javascripts/views/role.js]: Intentionally left as a hint how to provide a better solution.

File: app/assets/javascripts/views/role.js (hunk 3 above)
Comment (eleidan) [inline app/assets/javascripts/views/role.js]: See comments above.

Comment (afurm) [inline app/assets/javascripts/views/role.js]: :smiley:

**Pull request merged**